pkgs/development/guile-modules/guile-reader/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchurl, pkgconfig
+, lightning
+, gperf, guile, guile-lib, libffi }:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "guile-reader-${version}";
+  version = "0.6.2";
+
+  src = fetchurl {
+    url = "http://download.savannah.nongnu.org/releases/guile-reader/${name}.tar.gz";
+    sha256 = "0592s2s8ampqmqwilc4fvcild6rb9gy79di6vxv5kcdmv23abkgx";
+  };
+
+  patches = [ ./0001-fix-prototypes.diff ];
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ gperf guile guile-lib libffi ];
+
+  GUILE_SITE="${guile-lib}/share/guile/site";
+
+  preConfigure = ''
+    configureFlags="$configureFlags --with-guilemoduledir=$out/share/guile/site"
+  '';
+
+  doCheck = false;
```

Comment:
Please comment why the tests fail.

Reply:
With the new release of Lightning, they are working fine. I will update the code accordingly.